lib/internal/per_context/primordials.js

```diff
@@ -419,7 +421,13 @@ primordials.SafeWeakMap = makeSafe(
 primordials.SafeSet = makeSafe(
   Set,
   class SafeSet extends Set {
-    constructor(i) { super(i); } // eslint-disable-line no-useless-constructor
+    constructor(i) {
+      super(
+        i && ArrayIsArray(i) ?
```

Comment:
nit: the truthy test is probably superfluous. I can't find the implementation of `ArrayIsArray`, but it probably already does that, so this micro-op is likely actually a micro-deop.

Reply:
my assumption is that a truthiness check is much cheaper than any function call, but if a benchmark indicates that's untrue i'd be happy to change it in the future.